google/cloud/sql/connector/instance_connection_manager.py

```diff
@@ -343,17 +373,31 @@ async def _get_instance_data(self) -> InstanceMetadata:
             self._enable_iam_auth,
         )
 
-    def _auth_init(self) -> None:
+    def _auth_init(self, service_account_creds: Union[str, Credentials, None]) -> None:
         """Creates and assigns a Google Python API service object for
         Google Cloud SQL Admin API.
-        """
 
-        credentials, project = google.auth.default(
-            scopes=[
-                "https://www.googleapis.com/auth/sqlservice.admin",
-                "https://www.googleapis.com/auth/cloud-platform",
-            ]
-        )
+        :type service_account_creds: str | google.auth.credentials.Credentials | None
+        :param service_account_creds
+            Path to JSON service account key file to be used for authentication
+            or google.auth.credentials.Credentials object.
+            If not specified, Application Default Credentials are used.
+        """
+        scopes = [
+            "https://www.googleapis.com/auth/sqlservice.admin",
+            "https://www.googleapis.com/auth/cloud-platform",
+        ]
+        # if Credentials object is passed in, use for authentication
+        if isinstance(service_account_creds, Credentials):
+            credentials = with_scopes_if_required(service_account_creds, scopes=scopes)
+        # if string is passed in, load credentials from file
+        elif isinstance(service_account_creds, str):
+            credentials, project = google.auth.load_credentials_from_file(
+                filename=service_account_creds, scopes=scopes
+            )
```

Comment:
Hmm this seems pretty simple. Would we be better off just having the user do this and only taking in a `credentials` type?

Reply:
That works for me and makes it a little more straightforward to understand for users. I was just trying to mimic the go connector that allows credentials from JSON. I'll make it take in credentials, otherwise use default creds.

Reply:
That makes sense. I think for python, it would make more sense to either provide a helper function that loads the credential file with the correct scopes or just rescope the credentials after we get them.